### PR TITLE
Fix: tag manager settings stuck in loading state.

### PR DESCRIPTION
**Pull request by derweili**
## Summary

Addresses issue:

- #6464

## Relevant technical choices

* `setGAPropertyID` only when tagmanager settings not empty.
* Add tests for `useGAPropertyIDEffect` with empty tag manager settings.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/tagmanager/hooks/useGAPropertyIDEffect.js [hunk 1]
```diff
@@ -32,11 +32,21 @@ export default function useGAPropertyIDEffect() {
...
 	);
+	const tagmanagerExistingSettings = useSelect( ( select ) =>
+		select( MODULES_TAGMANAGER ).getSettings()
+	);
 	const { setGAPropertyID } = useDispatch( MODULES_TAGMANAGER );
 
 	useEffect( () => {
-		if ( singleAnalyticsPropertyID !== undefined ) {
+		if (
+			!! tagmanagerExistingSettings &&
```
Comment (tofumatt) [inline assets/js/modules/tagmanager/hooks/useGAPropertyIDEffect.js:42]: This is fine as once set, settings will be an object, but is there any reason not to use the slightly more explicit and clear `!== undefined` here? It just might make things a bit more readable, as that's the only non-truthy value we'd expect here, and our convention in code is that `undefined` means "loading"/"not yet loaded", so it makes it a bit more clear just from the condition that `tagmanagerExistingSettings` needs to be loaded, but otherwise the value doesn't matter.

```suggestion
			tagmanagerExistingSettings !== undefined &&
```

File: assets/js/modules/tagmanager/hooks/useGAPropertyIDEffect.test.js [hunk 2]
```diff
@@ -16,47 +16,102 @@
...
-		// Set settings to prevent fetch in resolver.
-		registry.dispatch( MODULES_TAGMANAGER ).receiveGetSettings( {} );
-		// Set set no existing tag.
-		registry.dispatch( MODULES_TAGMANAGER ).receiveGetExistingTag( null );
-	} );
 
-	it( 'sets the gaPropertyID with the current detected singular property ID in selected containers', async () => {
-		const { buildAndReceiveWebAndAMP } =
-			createBuildAndReceivers( registry );
+	describe( 'with empty tagmanager settings store', () => {
```
Comment (tofumatt) [inline assets/js/modules/tagmanager/hooks/useGAPropertyIDEffect.test.js:39]:
```suggestion
	describe( 'with empty Tag Manager settings store', () => {
```

File: assets/js/modules/tagmanager/hooks/useGAPropertyIDEffect.test.js [hunk 3]
```diff
@@ -16,47 +16,102 @@
...
+				)
+			);
 
-		buildAndReceiveWebAndAMP( {
-			webPropertyID: TEST_GA_PROPERTY_ID,
+			const propertyID = registry
+				.select( MODULES_TAGMANAGER )
+				.getGAPropertyID();
+
+			expect( propertyID ).toBe( '' );
```
Comment (tofumatt) [inline assets/js/modules/tagmanager/hooks/useGAPropertyIDEffect.test.js:75]: What exactly is this testing?

The test spec says it fetches from the API before setting/updating the value, but what was the value before? Why are we looking for an empty string for `propertyID`? I suspect it's because previously it was `undefined`, but because that isn't in this test anywhere it's hard-to-follow the logic here.

Could you expand on the test a bit, either with some comments highlighting what's being tested and happening "behind-the-scenes" of the selector calls, or with a few more `expect` statements?

File: assets/js/modules/tagmanager/hooks/useGAPropertyIDEffect.test.js [hunk 4]
```diff
@@ -16,47 +16,102 @@
...
+		} );
+	} );
+
+	describe( 'with tagmanager store setup', () => {
```
Comment (tofumatt) [inline assets/js/modules/tagmanager/hooks/useGAPropertyIDEffect.test.js:79]:
```suggestion
	describe( 'with a populated Tag Manager store', () => {
```

**Review by tofumatt — CHANGES_REQUESTED** [review]
Looks good, but the tests here could use a bit more explanation or maybe even just some renaming of test spec names.

One minor comment about the `!! tagmanagerExistingSettings` too, then we should be good-to-go 👍🏻

File: assets/js/modules/tagmanager/hooks/useGAPropertyIDEffect.test.js (hunk 3 above)
Comment (derweili) [reply assets/js/modules/tagmanager/hooks/useGAPropertyIDEffect.test.js]: You are right, this does not make sense. So I removed this part, as setting the propertyID is already tested in the other test, so we don't need it here.

So this test is now only verifies the settings are loaded from the API if the settings are empty. Do you think this makes sense?

**Review by tofumatt — APPROVED** [review]

File: assets/js/modules/tagmanager/hooks/useGAPropertyIDEffect.test.js (hunk 3 above)
Comment (tofumatt) [reply assets/js/modules/tagmanager/hooks/useGAPropertyIDEffect.test.js]: That makes a lot more senes, yeah. Thanks so much 👍🏻
